app/middlewares/validators/adminValidator.js

```diff
@@ -0,0 +1,19 @@
+const { User } = require('../../models/index');
+const errors = require('../../errors');
+const jwt = require('../../services/jwt');
+
+exports.validateAdmin = (req, res, next) => {
+  jwt.findUserByToken(req.headers.authorization).then(user => {
+    if (user.type !== 'admin') {
+      next(errors.authenticationError('User is not an Admin'));
+    }
+    next();
+  });
+};
+
+exports.create = (req, res, next) => {
+  User.findByPk(req.body.user_id).then(user => {
```

Comment:
Be careful with the databases errors handling, you're not caching nothing (this applies to several parts in this same PR)